src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -275,14 +278,46 @@ private boolean hasCramInput() {
     }
 
     /**
+     * The default implementation combines the default read filters for this tool (returned by
+     * {@link org.broadinstitute.hellbender.engine.ReadWalker#getDefaultReadFilter} with any read filter command
+     * line arguments specified by the user; wraps each filter in the resulting list with a CountingReadFilter;
+     * and returns a single composite filter resulting from the list by and'ing them together.
+     *
+     * Default tool implementation of {@link #traverse()} calls this method once before iterating
+     * over the reads and reuses the filter object to avoid object allocation. Nevertheless, keeping state in filter
+     * objects is strongly discouraged.
+     *
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter}
+     * composition methods.
+     */
+    /**
      * Returns the read filter (simple or composite) that will be applied to the reads returned from {@link #getReads}
-     * The default implementation uses the {@link org.broadinstitute.hellbender.engine.filters.WellformedReadFilter} filter with all default options.
+     * The default implementation combines the default read filters for this tool (returned by
+     * {@link org.broadinstitute.hellbender.engine.spark.GATKSparkTool#getDefaultReadFilters} with any read filter command
+     * line arguments specified by the user and returns a single composite filter resulting from the list by and'ing
+     * them together.
      *
-     * Subclasses can extend to provide their own filters (ie., override and optionally call super).
-     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter} composition methods.
+     * Multiple filters can be composed by using {@link org.broadinstitute.hellbender.engine.filters.ReadFilter}
+     * composition methods.
      */
     public ReadFilter makeReadFilter() {
-        return new WellformedReadFilter(getHeaderForReads());
+        final ReadFilterArgumentCollection rfc = readArguments.getReadFilterArgumentCollection();
+        return rfc.getMergedCommandLineFilters(getDefaultReadFilters())
+                .stream()
+                .map(f -> f.getFilterInstance(getHeaderForReads(), rfc))
+                .reduce(new WellformedReadFilter(getHeaderForReads()), (f1, f2) -> f1.and(f2));
```

Comment:
Code that looks like this shows up in a bunch of places.  Would it be possible to abstract it into ReadFilterArgumentCollection?  It looks like sometimes you need a CountingReadFilter and sometimes you need the read header, but is there any commonality that could be pulled up?